src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java

```diff
@@ -1122,6 +1129,22 @@ public void setClient(Client client) {
             this.mockClient = client;
         }
         
+        @Override
+        protected Invocation buildClientRequest(String url, Collection<WebServiceCredential> creds) throws URISyntaxException {
+            if (StringUtils.equals(url, 
+                    "https://www.cornell.edu/v1_0/invoice?year=2019&month=09Sep&filename=J%20Aron%20&%20Company%20B34367.pdf")) {
+                return allowClientToFailWithPoorlyFormatedUrl(url);
+            } else {
+                return super.buildClientRequest(url, creds);
+            }
+        }
+
+        private Invocation allowClientToFailWithPoorlyFormatedUrl(String url) throws URISyntaxException {

```

Comment:
Could you help me understand why this method is needed?

Reply:
Whem the bad attachment URL example is passed to the function call "super.getClient().target(uri).request()", a UriBuilderException is thrown.  I wanted to create this scenario like we see in the batch job.  I am guessing a better function name would help here.  Do you have a recommendation?

Reply:
When this error happens in production is it being thrown by the same code in buildClientRequest? I still don't understand why we had to override the buildClientRequest implementation

Reply:
The client is set to a mock client that is useful for the other tests.  THe client was needed for this test so that it would throw the real exception.
And Yes, this where it happening in production I believe

Reply:
I understand now. I think changing this method's name would make sense. Something that would explain it is not using the mock client.